lib/rubocop/cop/rspec/memoized_helper_block_delimiter.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Use consistent style block delimiters for memoized helpers.
+      #
+      # @example EnforcedStyle: do_end (default)
+      #   # bad
+      #   let(:foo) { 'bar' }
+      #
+      #   # good
+      #   let(:foo) do
+      #     'bar'
+      #   end
+      #
+      #   # bad
+      #   subject(:foo) { 'bar' }
+      #
+      #   # good
+      #   subject(:foo) do
+      #     'bar'
+      #   end
+      #
+      # @example EnforcedStyle: braces
+      #   # bad
+      #   let(:foo) do
+      #     'bar'
+      #   end
+      #
+      #   # good
+      #   let(:foo) { 'bar' }
+      #
+      #   # bad
+      #   subject(:foo) do
+      #     'bar'
+      #   end
+      #
+      #   # good
+      #   subject(:foo) { 'bar' }
+      class MemoizedHelperBlockDelimiter < Base
+        extend AutoCorrector
+
+        include ConfigurableEnforcedStyle
+        include RangeHelp
+        include RuboCop::RSpec::Language
+
+        # @param node [RuboCop::AST::BlockNode]
+        # @return [void]
+        def on_block(node) # rubocop:disable InternalAffairs/NumblockHandler
+          return unless bad?(node)
+
+          add_offense(
+            node.location.begin.with(
+              end_pos: node.location.end.end_pos
+            ),
+            message: "Use #{style} style block delimiters."
+          ) do |corrector|
+            autocorrect(corrector, node)
+          end
+        end
+
+        private
+
+        # @param corrector [RuboCop::Cop::Corrector]
+        # @param node [RuboCop::AST::BlockNode]
+        # @return [void]
+        def autocorrect(corrector, node)
+          case style
+          when :braces
+            # Not supported because it conflicts with Style/BlockDelimiters.
+          when :do_end
+            autocorrect_braces(corrector, node)
+          end
+        end
+
+        # @param corrector [RuboCop::Cop::Corrector]
+        # @param node [RuboCop::AST::BlockNode]
+        # @return [void]
+        def autocorrect_braces(corrector, node)
+          unless whitespace_before?(node.location.begin)
+            corrector.insert_before(node.location.begin, ' ')
+          end
+          corrector.replace(node.location.begin, 'do')
+          corrector.replace(node.location.end, 'end')
+          wrap_in_newlines(corrector, node) if node.single_line?
+        end
+
+        # @param node [RuboCop::AST::BlockNode]
+        # @return [Boolean]
+        def bad?(node)
+          memoized_helper_method?(node) &&
```

Comment:
This line doesn't semantically belong to "bad", it detects if it's a match.
I suggest moving it to `on_block`:
```ruby
memoized_helper_method(node) do |memoized_helper|
  next if preferred_block_delimiter?(memoized_helper)
  add_offense(...)
```